tests/pool_alloc.c

```diff
@@ -0,0 +1,70 @@
+#include <pool_alloc.h>
+#include <stdc.h>
+#include <malloc.h>
+#include <ktest.h>
+
+void int_constr(void *buf, __attribute__((unused)) size_t size) {
+  int *num = buf;
+  *num = 0;
+}
+void int_destr(__attribute__((unused)) void *buf,
+               __attribute__((unused)) size_t size) {
+}
+
+void test_pool_alloc(
+  int flag) { /* 0 - regular test, 1 - memory corrruption, 2 - double free */
```

Comment:
I know, I know, this is just a simple test - but these flags would be much simpler to read and understand if you made them a `#define` or `enum`.